source/extensions/common/wasm/wavm/wavm.h

```diff
@@ -13,12 +13,13 @@
 #include "common/common/c_smart_ptr.h"
 #include "common/common/logger.h"
 
-#include "extensions/common/wasm/wasm.h"
-
 namespace Envoy {
 namespace Extensions {
 namespace Common {
 namespace Wasm {
+
+class WasmVm;
+
```

Comment:
Trading header inclusion for forward declaration is not a best idea, since you get an incomplete type.

Reply:
+1

Reply:
We only need incomplete type because the only usage of WasmVm is "unique_ptr<WasmVm>".

Reply:
envoy doesn't use forward declaration unless there is cycle dependency. I will revert this change.